converters/fromZigbee.js

```diff
@@ -5556,6 +5556,53 @@ const converters = {
             return result;
         },
     },
+    schneider_ui_action: {
+        cluster: 'wiserDeviceInfo',
+        type: 'attributeReport',
+        convert: (model, msg, publish, options, meta) => {
+            if (hasAlreadyProcessedMessage(msg)) return;
+
+            const data = msg.data['deviceInfo'].split(',');
+            if(data[0] === 'UI' && data[1]) {
+                publish({action: data[1]});
+
+                let screenAwake = globalStore.getValue(msg.endpoint, 'screenAwake');
+                screenAwake = screenAwake != undefined ? screenAwake : false;
+                let keypadLocked = msg.endpoint.getClusterAttributeValue('hvacUserInterfaceCfg', 'keypadLockout');
+                keypadLocked = keypadLocked != undefined ? keypadLocked != 0 : false;
+
+                // Emulate UI temperature update
+                if (data[1] === 'ScreenWake') {
+                    globalStore.putValue(msg.endpoint, 'screenAwake', true);
+                } else if (data[1] === 'ScreenSleep') {
+                    globalStore.putValue(msg.endpoint, 'screenAwake', false);
+                } else if (screenAwake && !keypadLocked) {
```

Comment:
Note that the `screenAwake` is not reliable (because when z2m restarts this value resets). I'm wondering why this is needed, is this to prevent the occupiedHeatingSetpoint from being decreased when the screen is sleeping?

Reply:
When the device is awoken by button press, the button pressed event is sent by the device but it should not change the setpoint.
Note: The device computes and  displays the updated setpoint locally, then reads the actual setpoint back from the coordinator after a few seconds. For good user experience, the behavior of the device should be matched as closely as possible.

Reply:
Fixed